src/python/zquantum/core/history/recorder.py

```diff
@@ -1,6 +1,6 @@
 """Main implementation of the recorder."""
 import copy
-from typing import Any, Callable, Dict, Generic, List, NamedTuple, TypeVar
+from typing import Any, Callable, Dict, Generic, List, NamedTuple, TypeVar, Union
```

Comment:
As far as I can tell this `Union` is unnecessary here.
```suggestion
from typing import Any, Callable, Dict, Generic, List, NamedTuple, TypeVar
```

Reply:
fixed in c7f962badf69175fbfdcfca8cb5d226bd5bbd78b